pkg/lib/cockpit-components-terminal.jsx

```diff
@@ -130,6 +138,37 @@ export class Terminal extends React.Component {
         window.removeEventListener('resize', this.onWindowResize);
     }
 
+    setText() {
+        try {
+            navigator.clipboard.readText()
+                    .then((text) => {
+                        this.props.channel.send(text);
+                    })
+                    .catch(() => {
+                        console.error('Text could not be pasted, use Shift+Insert');
+                    })
+                    .finally(() => {
+                        this.state.terminal.focus();
+                    });
+        } catch (error) {
+            console.error('Text could not be pasted, use Shift+Insert');
```

Comment:
Why does this need to be caught twice?

Reply:
Answer to myself: I've seen this fail on Firefox with the "outer" exception, as the method call fails due  to the method not existing. But the returned promise fails in chromium if it doesn't have permissions, so it really can fail both ways.